### PR TITLE
refactor(share/eds): add ErrNotFound to eds store and shrex server

### PR DESCRIPTION
**Pull request by walldiss**
## Overview
If item doesn't exist in store, return explicit error `ErrNotFound`.

### CODE REVIEW

File: share/eds/store.go [hunk 1]
```diff
@@ -36,6 +37,8 @@ const (
 	defaultGCInterval = time.Hour
 )
 
+var ErrItemNotExist = errors.New("item not exist in store")
```
Comment (renaynay) [inline share/eds/store.go:40]:
```suggestion
var ErrItemNotExist = errors.New("item does not exist in store")
```

**Review by renaynay — COMMENTED** [review]
I like checking for the shard not exist part and responding with NOT FOUND, but I don't think the error wrapper is necessary - can we just check `if errors.Is(err, dagstore.ErrShardUnknown) ` instead?

**Comment by walldiss** [comment]
error wrapper is made to clean up dependancy graph. So shrex server doesn't rely on internal eds store implementation dependancies.

File: share/eds/store.go [hunk 2]
```diff
@@ -36,6 +37,8 @@ const (
 	defaultGCInterval = time.Hour
 )
 
+var ErrItemNotExist = errors.New("item does not exist in store")
```
Comment (Wondertan) [inline share/eds/store.go:40]:
```suggestion
var ErrItemNotExist = errors.New("eds: item does not exist in store")
```

**Review by Wondertan — COMMENTED** [review]
Why do you add item? It's only about eds, so we can omit item. I would say item is even confusing as its not immediately clear that item means eds.

We also use NotFound everywhere wording rather then NotExist so let's make it consistent

File: share/eds/store.go (hunk 2 above)
Comment (renaynay) [reply share/eds/store.go]: fine w me, let's just say 
`eds: not found in store`

**Review by Wondertan — APPROVED** [review]

**Comment by Wondertan** [comment]
Also we need to update pR title and description

**Review by distractedm1nd — APPROVED** [review]

**Pull request merged**